snappy/snapp.go

```diff
@@ -849,21 +849,21 @@ func (s *SnapPart) Install(inter progress.Meter, flags InstallFlags) (name strin
 		return "", err
 	}
 
-	// and finally make active
-	err = s.activate(inhibitHooks, inter)
-	defer func() {
-		if err != nil && oldPart != nil {
-			if cerr := oldPart.activate(inhibitHooks, inter); cerr != nil {
-				logger.Noticef("When setting old %s version back to active: %v", s.Name(), cerr)
+	if !inhibitHooks {
+		// and finally make active
+		err = s.activate(inhibitHooks, inter)
+		defer func() {
+			if err != nil && oldPart != nil {
+				if cerr := oldPart.activate(inhibitHooks, inter); cerr != nil {
```

Comment:
Looking above makes me think there are likely other such issues in Install. The TODO/bug should probably be more generic than this one case, and encompass a review of Install/Remove themselves.